src/OAuth/Credential.php

```diff
@@ -92,4 +92,20 @@ public function getRefreshToken(): string
     {
         return $this->refreshToken;
     }
+
+    /**
+     * @param string $accessToken
+     * @param string $refreshToken
+     *
+     * @return Credential
+     */
+    public function withRefreshedTokens(string $accessToken, string $refreshToken): self
```

Comment:
`withNewCredentials` is not good either, since "credential" means (LS AccountID + access token + refresh token) here. Maybe `withRefreshedToken()` sounds more clear?